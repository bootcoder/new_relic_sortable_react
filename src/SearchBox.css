.info-box {
  background-color: #5cb85c;
}

.select-margin-top {
  margin-top: 7px;
}

.margin-5p {
  margin: 5px 0;
}

@media (min-width: 1200px) {
    .pull-right-lg {
        float: right;
    }
}
