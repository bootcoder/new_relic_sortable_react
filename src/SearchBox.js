import React, { Component } from 'react'
import PropTypes from 'prop-types'
import './SearchBox.css'

class SearchBox extends Component {
  render () {
    return (
      <div className='SearchBox'>
        <input id='name-input' type='text' defaultValue={this.props.name} onChange={this.props.handleUpdateName} />
      </div>
    )
  }
}

SearchBox.propTypes = {
  name: PropTypes.string.isRequired,
  company: PropTypes.string.isRequired,
  sortBy: PropTypes.string.isRequired,
  handleUpdateName: PropTypes.func.isRequired
}

export default SearchBox
