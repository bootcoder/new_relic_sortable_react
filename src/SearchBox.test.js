/* global describe, it, ,sinon, expect, beforeEach, mount, shallow */

import SearchBox from './SearchBox'

describe('<SearchBox />', () => {
  let wrapper
  let nameSpy
  beforeEach(() => {
    nameSpy = sinon.spy()
    wrapper = mount(
      <SearchBox
        company='New Relic'
        name='Fred'
        sortBy='last_name_ascn'
        handleUpdateName={nameSpy} />)
  })

  it('renders without crashing', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('has param props', () => {
    expect(wrapper.props().company).toEqual('New Relic')
    expect(wrapper.props().name).toEqual('Fred')
    expect(wrapper.props().sortBy).toEqual('last_name_ascn')
  })

  describe('name input', () => {
    let nameInput
    beforeEach(() => {
      nameInput = wrapper.find('#name-input')
    })

    it('has a text input for name', () => {
      expect(nameInput.length).toEqual(1)
    })

    it('has the name param as input value', () => {
      expect(nameInput.get(0).props.defaultValue).toEqual('Fred')
    })

    it('updates the name field onChange', () => {
      nameInput.simulate('change', {target: {value: 'Fredi'}})
      expect(nameSpy.calledOnce).toBe(true)
    })
  })
})
