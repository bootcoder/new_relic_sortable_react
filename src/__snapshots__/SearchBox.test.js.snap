// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBox /> renders without crashing 1`] = `
<div
  className="SearchBox"
>
  <input
    defaultValue="Fred"
    id="name-input"
    onChange={[Function]}
    type="text"
  />
</div>
`;
